scipy/optimize/_minimize.py

```diff
@@ -514,6 +514,14 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
 
     """
     x0 = np.atleast_1d(np.asarray(x0))
+
+    if x0.ndim != 1:
+        message = ('Use of `minimize` with `x0.ndim != 1` is deprecated. '
+                   'Currently, singleton dimensions will be removed from '
+                   '`x0`, but an error may be raised in the future.')
+        warn(message, DeprecationWarning)
```

Comment:
does this need a stacklevel?

Reply:
Yes but difficult to say how much. I would guess 4 as usually called from a private optimization function.

Reply:
This is right in `minimize`, which I think we should assume is being called from the command line or a script. Wouldn't it be `stacklevel=2`, then?

Reply:
I'm calling minimize on line 26 of a script:
![image](https://user-images.githubusercontent.com/6570539/147629942-d147229e-65ea-4a3b-b9d1-ee12393cd475.png)

With `stacklevel=2`, here is the `DeprecationWarning`:
```
C:\Users\matth\Dropbox\Scipy\Bounds\bounds_test.py:26: DeprecationWarning: Use of `minimize` with `x0.ndim != 1` is deprecated. Currently, singleton dimensions will be removed from `x0`, but an error may be raised in the future.
  optimize.minimize(lambda x: np.linalg.norm(x), x0=np.matrix([1, 2]), method='cobyla')
```

Reply:
Added `stacklevel=2`.

Reply:
Oh right, I thought we where in the Bounds here my bad. 2 is correct yes.